x/staking/types/msg_test.go

```diff
@@ -14,6 +17,25 @@ var (
 	coinZero = sdk.NewInt64Coin(sdk.DefaultBondDenom, 0)
 )
 
+func TestMsgPkDecode(t *testing.T) {
+	// description := Description{}
+	// commission1 := NewCommissionRates(sdk.ZeroDec(), sdk.ZeroDec(), sdk.ZeroDec())
+	// msg, err := NewMsgCreateValidator(valAddr1, pk1, coinPos, description, commission1, sdk.OneInt())
+	// require.NoError(t, err)
+
+	registry := codectypes.NewInterfaceRegistry()
+	cdc := codec.NewProtoCodec(registry)
+
+	pk1bz, err := codec.MarshalAny(cdc, pk1)
+	require.NoError(t, err)
+
+	var pkUnmarshaled ed25519.PubKey
```

Comment:
You need to pass `*cryptotypes.PubKey` interface (not the concrete `*ed25519.PubKey` type) to `UnmarshalAny`:

```suggestion
	var pkUnmarshaled cryptotypes.PubKey
```